src/vmspawn/vmspawn.c

```diff
@@ -438,6 +454,12 @@ static int parse_argv(int argc, char *argv[]) {
 
                         break;
 
+                case ARG_BACKGROUND:
+                        r = free_and_strdup_warn(&arg_background, optarg);
```

Comment:
Just use optarg (there's no need to strdup() this?)

Reply:
So, we probably should make it a rule to always copy such values. At multiple occasions before this led to very very confusing bugs given that in safe_fork() we use rename_process() which patches around in argv[] memory to rename a process for "ps". But if we do that, then any pointers into argv[] (such as one derived from optarg) will be invalidated.

Hence, unless the tool is really simple we probably should always strdup() strings like this, because the minor speed improvement really isn't worth it given how nasty and irritating it is to debug the rename_process() mess afterwards...

Reply:
OK. Make sense.